src/main/java/jsr223/kubernetes/KubernetesScriptEngine.java

```diff
@@ -347,6 +358,10 @@ private void populateBindingAndGI() throws ScriptException {
                 k8sResourceToStream = genericInfo.get("K8S_RESOURCE_TO_STREAM");
             }
         }
+
+        // Retrieving passed variables if any
+        variables = (Map<String, String>) context.getBindings(ScriptContext.ENGINE_SCOPE)
+                                                 .get(SchedulerConstants.VARIABLES_BINDING_NAME);
```

Comment:
instead of only getting the variables and generic information bindings, it is more generic to simply iterate existing bindings and depending on the type, build a replacement map. 
Example:
1) if binding is a map:
for each key inside the map, ${bindingName_key} => value.toString()
2) if binding is a list, array, collection(?)
for each element, ${bindingName_index} => value.toString()
3) otherwise
${bindingName} => value.toString()

Reply:
ref https://github.com/ow2-proactive/jsr223-nativeshell/blob/master/src/main/java/jsr223/nativeshell/NativeShellRunner.java